Require Developer Certificate of Origin in pull requests (#15613)
PowerDNS/pdns: CONTRIBUTING.md

* The rest of the commit body should be wrapped at 72 characters (see [this](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html) for more info)
* If this commit fixes an issue, put "Closes #XXXX" in the message
* Do not put whitespace fixes/cleanup and functionality changes in the same commit

> * Include a valid "Signed-off-by" line as a [Developer Certificate of Origin](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin). Our DCO can be found [here](https://github.com/PowerDNS/pdns/blob/master/DCO)

Reviewer: Thus maybe
```suggestion
* Include a valid "Signed-off-by" line as a [Developer Certificate of Origin](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin). Our DCO can be found [on the PowerDNS website](https://github.com/PowerDNS/pdns/blob/master/DCO)
```
will do.

Reviewer 2: Not great. Links shouldn't say where you can find things, they should say what you can find.

Reviewer 2: ```suggestion
* Include a valid "Signed-off-by" line as a [Developer Certificate of Origin](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin). See the [PowerDNS DCO](https://github.com/PowerDNS/pdns/blob/master/DCO).
```

# Developer Certificate of Origin

> We require a "Signed-Off" on all commits contributed to the PowerDNS codebase, as a [Developer Certificate of Origin](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin), version [1.1](https://github.com/PowerDNS/pdns/blob/master/DCO)

Reviewer 3: same here

> We require a "Signed-off-by" on all commits contributed to the PowerDNS codebase, as a [Developer Certificate of Origin](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin), version [1.1](https://github.com/PowerDNS/pdns/blob/master/DCO)

Reviewer 2: This should really have a "as of _some release_" statement, as it is not a statement about the entirety of commits to the repository.

Author: I'm not sure mentioning a release here makes sense:
- we will start requiring SOB for contributions to any branches as soon as this PR is merged
- people reading this document are willing to contribute new code, and usually don't care about what we did in the past. The document lists a lot of steps (code formatting, spelling, static analyzers, ...) that are now required but weren't in the past.
